ssl/ssl_test.go

```diff
@@ -315,6 +398,27 @@ func TestCreateCA(t *testing.T) {
 
 		assert.NotEqual(t, secrets.Data[certInfo[certID].PrivateKeyName], []byte{})
 		assert.NotEqual(t, secrets.Data[certInfo[certID].CertificateName], []byte{})
+		assert.Regexp(t, "(?s)\\A-----BEGIN CERTIFICATE-----\\n.*\\n-----END CERTIFICATE-----\\n\\z",
+			string(secrets.Data[certInfo[certID].CertificateName]))
+	})
+
+	t.Run("createCA appends kube CA cert when requested", func(t *testing.T) {
+		t.Parallel()
+
+		certInfo := make(map[string]CertInfo)
+		certInfo[certID] = CertInfo{
+			PrivateKeyName:  "private-key",
+			CertificateName: "certificate-name",
+			KubeCACertFile:  "testdata/kubeca.crt",
+		}
+		secrets := &v1.Secret{Data: map[string][]byte{}}
+
+		createCA(certInfo, secrets, certID, 365)
+
+		assert.NotEqual(t, secrets.Data[certInfo[certID].PrivateKeyName], []byte{})
+		assert.NotEqual(t, secrets.Data[certInfo[certID].CertificateName], []byte{})
+		assert.Contains(t, string(secrets.Data[certInfo[certID].CertificateName]),
+			"-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----")
```

Comment:
Shouldn't we test that the content of `kubeca.crt` was inserted (and not, say, two copies of the internal CA)?

Reply:
Fixed in 8cf5338